src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/DependenciesProjectTreeProvider.cs

```diff
@@ -0,0 +1,973 @@
+                                                     IDependenciesGraphProjectContext
+    {
+        private static readonly ProjectTreeFlags DependenciesRootNodeFlags
+                = ProjectTreeFlags.Create(ProjectTreeFlags.Common.BubbleUp
+                                          | ProjectTreeFlags.Common.ReferencesFolder
+                                          | ProjectTreeFlags.Common.VirtualFolder)
+                                  .Add("DependenciesRootNode");
+
+        // TODO: Consider moving this to an [AppliesTo] attribute and
+        // removing the GetIsApplicableAsync method or always returning true from it.
+        /// <summary>
+        /// The AppliesTo expression that must be satisfied for this subtree provider to be applicable.
+        /// </summary>
+        private static readonly string ApplicabilityTest 
+                = string.Format(CultureInfo.InvariantCulture, "{0} & ({1} | {2} | {3} | {4} | {5})",
+                                ProjectCapabilities.ReferencesFolder,
+                                ProjectCapabilities.AssemblyReferences,
+                                ProjectCapabilities.ComReferences,
+                                ProjectCapabilities.ProjectReferences,
+                                ProjectCapabilities.SdkReferences,
+                                ProjectCapabilities.WinRTReferences);
+
+        /// <summary>
+        /// Keeps latest updated snapshot of all rules schema catalogs
+        /// </summary>
+        private IImmutableDictionary<string, IPropertyPagesCatalog> NamedCatalogs { get; set; }
+
+        /// <summary>
+        /// Gets the collection of <see cref="IProjectTreePropertiesProvider"/> imports 
+        /// that apply to the references tree.
+        /// </summary>
+        [ImportMany(ReferencesProjectTreeCustomizablePropertyValues.ContractName)]
+        private OrderPrecedenceImportCollection<IProjectTreePropertiesProvider> ProjectTreePropertiesProviders { get; set; }
+
+        /// <summary>
+        /// Provides a way to extend Dependencies node by consuming sub tree providers that represent
+        /// a particular dependency type and are responsible for Dependencies\[TypeNode] and it's contents.
+        /// </summary>
+        [ImportMany]
+        private OrderPrecedenceImportCollection<IProjectDependenciesSubTreeProvider> SubTreeProviders { get; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DependenciesProjectTreeProvider"/> class.
+        /// </summary>
+        [ImportingConstructor]
+        public DependenciesProjectTreeProvider(IProjectThreadingService threadingService, 
+                                               UnconfiguredProject unconfiguredProject)
+            : base(threadingService, unconfiguredProject)
+        {
+            ProjectTreePropertiesProviders = new OrderPrecedenceImportCollection<IProjectTreePropertiesProvider>(
```

Comment:
this is how OrderPrecedenceImportCollection works.

---
In reply to: [74317181](https://github.com/dotnet/roslyn-project-system/pull/349#discussion_r74317181) [](ancestors = 74317181)